manager.py

```diff
@@ -0,0 +1,160 @@
+import docker
+import psycopg2
+import json
+import time
+
+
+def get_env_vars(client):
+    global postgres_user
+    global postgres_pass
+    global postgres_db
+
+    container = client.containers.get("citus_master")
+
+    env_vars = container.exec_run(cmd='env')
+
+    for each_line in env_vars.splitlines():
+        if 'POSTGRES' in each_line:
+            (key, val) = each_line.split("=")
+            if key == "POSTGRES_USER":
+                postgres_user = val
+            elif key == "POSTGRES_PASSWORD":
+                postgres_pass = val
+            elif key == "POSTGRES_DB":
+                postgres_db = val
+
+
+def connect_to_db(client):
+    conn = None
+    a = 0
+
+    get_env_vars(client)
+
+    while a < 10 and conn is None:
+        time.sleep(2)
+        try:
+            conn = psycopg2.connect("dbname=%s user=%s host=%s password=%s" %
+                                    (postgres_db, postgres_user, "citus_master", postgres_pass))
+        except:
+            print("I am unable to connect")
+            a = a + 1
+
+    return conn
+
+
+def add_workers(conn, workers):
+    for worker in workers:
+        add_worker(conn, worker, 5432)
+    conn.commit()
+
+
+def add_worker(conn, host, port):
+    cur = conn.cursor()
+    worker_dict = ({"host": host, "port": port})
+    try:
+        cur.execute("""SELECT master_add_node(%(host)s, %(port)s)""", worker_dict)
+    except:
+        print("I can't add worker to the coordinator!")
+
+
+def remove_worker(conn, host, port):
```

Comment:
This was the part I was playing around with yesterday and I realized it has to go. The dropping of machines when they are scaled down doesn't work correctly with our shard health. Better to just leave them in and let the master mark their shards as unhealthy during modifications, which is to say we can just remove the `remove_worker` logic altogether.

Reply:
Currently, if we see a die command, the manager sends a remove query. I know that if there is a shard in the worker to be removed, master rejects the query. However, when I was writing that method, the idea was to provide the ability to remove a worker if we never used it. Let's assume the user was using a cluster with 2 workers and at some point he scaled-up to 4 workers and without even putting any data he scaled-down to 3 workers. If we do not keep remove_worker method, the cluster becomes useless because the user will not be able to create any distributed tables. However, if we were to keep it, he could have continued to use it. 

AFAIK the previous version of workerlist-gen also allows to scale-down and master simply rejects if there is data. 

This might not be an appropriate use-case but on the other hand, I believe the method is harmless.